ivy/functional/frontends/tensorflow/linalg.py

```diff
@@ -253,3 +253,8 @@ def einsum(equation, *inputs, **kwargs):
 @to_ivy_arrays_and_back
 def adjoint(matrix, name=None):
     return ivy.adjoint(matrix)
+
+
+@to_ivy_arrays_and_back
```

Comment:
Hey @mobley-trent 

The function signature you used here is incorrect I guess given the documentation of this function it indicates that it should be defined as 
```python
def diag(
    diagonal,
    name='diag',
    k=0,
    num_rows=-1,
    num_cols=-1,
    padding_value=0,
    align='RIGHT_LEFT'
):
    pass
```
You can read more here https://www.tensorflow.org/api_docs/python/tf/linalg/diag